### ISSUE TITLE
Crash if tap on a card just after the release scroll. 

### ISSUE THREAD

**Issue by VinayMaheshwari** [gleue/TGLStackedViewController#37]
Tap on a card to see exposed view. Scroll up down (Pan gesture) in a way the [self.collectionView cancelInteractiveTransition]; been called from handleCollapsePanGesture: case UIGestureRecognizerStateEnded. Just a that movement tap on the exposed card. App crashes with the below logs.

> 2016-08-03 19:25:03.766 TGLStackedViewExample[358:20875] -[TGLStackedLayout setTransitionProgress:]: unrecognized selector sent to instance 0x125f176f0
> 2016-08-03 19:25:03.768 TGLStackedViewExample[358:20875] **\* Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[TGLStackedLayout setTransitionProgress:]: unrecognized selector sent to instance 0x125f176f0'
> **\* First throw call stack:
> (0x180b76db0 0x1801dbf80 0x180b7dc4c 0x180b7abec 0x180a78c5c 0x18651d538 0x18651dfe8 0x185f436d0 0x185f43434 0x18154d038 0x180b2d794 0x180b2d438 0x180b2ab4c 0x180a54c50 0x18233c088 0x185d3e088 0x10008390c 0x1805f28b8)
> libc++abi.dylib: terminating with uncaught exception of type NSException

**Comment by VinayMaheshwari** [comment]
NOTE: I have disabled the MoveGesture. Because I don't need the rearrange feature.
 //[self.collectionView addGestureRecognizer:self.moveGestureRecognizer];

**Comment by carlmknott** [comment]
Yes, I get the same problem. Any ideas? PS This project is AWESOME.

**Comment by gleue** [comment]
Sorry for not responding -- been on vacation. Will be back to my desk on monday ;)

**Comment by VinayMaheshwari** [comment]
I am still able to reproduce the crash.

Device logs:

> Aug 31 13:14:25 iPhone SpringBoard[58] <Error>:  SecTrustEvaluate  [leaf IssuerCommonName SubjectCommonName]
> Aug 31 13:14:25 iPhone SpringBoard[58] <Error>:  SecTrustEvaluate  [leaf IssuerCommonName SubjectCommonName]
> Aug 31 13:14:25 iPhone kernel[0] <Notice>: xpcproxy[4250] Container: /private/var/mobile/Containers/Data/Application/49C479C5-1298-401E-8243-C161E3ADFD7B (sandbox)
> Aug 31 13:14:25 iPhone com.apple.xpc.launchd[1] <Error>: assertion failed: 13F69: launchd + 116796 [9F6284CF-8A17-36CC-9DB5-85D510A21F14]: 0x3
> Aug 31 13:14:25 iPhone SpringBoard[58] <Warning>: LICreateIconForImage passed NULL CGImageRef image
> Aug 31 13:15:07 iPhone TGLStackedViewExample[4250] <Error>: -[TGLStackedLayout setTransitionProgress:]: unrecognized selector sent to instance 0x1266ac2a0
> Aug 31 13:15:07 iPhone TGLStackedViewExample[4250] <Error>: *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[TGLStackedLayout setTransitionProgress:]: unrecognized selector sent to instance 0x1266ac2a0'
> 	*** First throw call stack:
> 	(0x18145edb0 0x180ac3f80 0x181465c4c 0x181462bec 0x181360c5c 0x186e05538 0x186e05fe8 0x18682b6d0 0x18682b434 0x181e35038 0x181415794 0x181415438 0x181412b4c 0x18133cc50 0x182c24088 0x186626088 0x1000bb83c 0x180eda8b8)
> Aug 31 13:15:07 iPhone SpringBoard[58] <Warning>: HW kbd: Failed to set (null) as keyboard focus
> Aug 31 13:15:07 iPhone com.apple.xpc.launchd[1] (UIKitApplication:com.gleue-interactive.TGLStackedViewExample[0xaf05][4250]) <Notice>: Service exited due to signal: Abort trap: 6
> Aug 31 13:15:07 iPhone diagnosticd[85] <Error>: unable to find offset 0x80ecf98c in shared cache for arch 'arm64'
> Aug 31 13:15:07 iPhone ReportCrash[4251] <Notice>: platform_thread_get_unique_id matched 1201426
> Aug 31 13:15:07 iPhone ReportCrash[4251] <Notice>: Formulating report for corpse[4250] TGLStackedViewExample
> Aug 31 13:15:07 iPhone ReportCrash[4251] <Warning>: Saved type '109(109_TGLStackedViewExample)' report (2 of max 25) at /var/mobile/Library/Logs/CrashReporter/TGLStackedViewExample-2016-08-31-131507.ips
> Aug 31 13:15:07 iPhone SpringBoard[58] <Warning>: Application 'UIKitApplication:com.gleue-interactive.TGLStackedViewExample[0xaf05]' crashed.
> Aug 31 13:15:07 iPhone UserEventAgent[26] <Warning>: 8738172920852: id=com.gleue-interactive.TGLStackedViewExample pid=4250, state=0
> Aug 31 13:15:07 iPhone searchd[155] <Warning>: ====^^^^ DuetExpert missing data, count -> people:0 applicationDeepLinks:0 applications:8 requests:158 missingAllDataRequests:0
> Aug 31 13:15:39 iPhone syncdefaultsd[4252] <Notice>: (Note ) SYDAccount: no account
> Aug 31 13:15:39 iPhone syncdefaultsd[4252] <Notice>: (Note ) SYDBootAccount: no account (null)
> Aug 31 13:15:39 iPhone syncdefaultsd[4252] <Notice>: (Note ) SYDPIMAccount: no account (null)
> Aug 31 13:15:39 iPhone syncdefaultsd[4252] <Notice>: (Note ) SYDAlwaysOnAccount: no account (null)

NOTE: In  code i just comment out 
// [self.collectionView addGestureRecognizer:self.moveGestureRecognizer];
because I do not want the move gesture.

Attached the video (Please ignore -Audio) to show how to reproduce crash.
[IMG_0209.MOV.zip](https://github.com/gleue/TGLStackedViewController/files/446680/IMG_0209.MOV.zip)

**Comment by VinayMaheshwari** [comment]
As per my understanding  with your recent changes "addition of interactiveTransitionInProgress flag" not allowing selection of card while Transition In Progress. 
But it may be case of cancelInteractiveTransition.

**Comment by gleue** [comment]
Ok, I had to try hard as in your video, but I could reproduce the crash. I'll have to investigate further, I guess

**Comment by gleue** [comment]
Could you please test again w/ release 2.1.2?

**Comment by VinayMaheshwari** [comment]
Sure, I will update you by Monday. Thank you.

**Comment by gleue** [comment]
Fixed another bug regarding interactive transitions in 2.1.3

**Comment by abdultcb** [comment]
Thanks buddy.

**Comment by kiritoxcii** [comment]
This should be open, I'm having the same crash constantly in version 2.2
